plugin/core/workspace.py

```diff
@@ -56,9 +58,32 @@ class ProjectFolders(object):
     def __init__(self, window: sublime.Window) -> None:
         self._window = window
         self.folders = self._window.folders()  # type: List[str]
+        # Per-folder ignore patterns. The list order matches the order of self.folders.
+        self._folders_exclude_patterns = []  # type: List[List[str]]
+        self._update_exclude_patterns(self.folders)
+
+    def _update_exclude_patterns(self, folders: List[str]) -> None:
+        self._folders_exclude_patterns = []
+        project_data = self._window.project_data()
+        if not isinstance(project_data, dict):
+            return
+        for i, folder in enumerate(project_data.get('folders', [])):
+            exclude_patterns = []
+            self._folders_exclude_patterns.append(exclude_patterns)
+            # Use canoncial path from `window.folders` rather than potentially relative path from project data.
+            path = folders[i]
```

Comment:
I haven't tested it, but what is this `folders` variable? Should it be `self.folders`?

Also could you move the `self._folders_exclude_patterns.append(exclude_patterns)` line to the end; I'm always unsure for which data structures Python treats variables by reference if you modify them later, and reordering those lines would make the logic clearer.

Reply:
`folders` is `self.folders` but it's passed as an argument to this function rather than accessed directly as I feel that improves the clarity in regards to the timing/order of when `self.folders` is updated relative to these patterns being update. In other words, if this code would access `self.folders` directly then I feel it would be easier to make a mistake of running this code before `self.folders` is updated.